src/compiler/base/AwaitableNode.ts

```diff
@@ -0,0 +1,95 @@
+import * as ts from "typescript";
+import {Constructor} from "./../../Constructor";
+import * as errors from "./../../errors";
+import {AwaitableNodeStructure} from "./../../structures";
+import {callBaseFill} from "./../callBaseFill";
+import {insertIntoParent, removeChildrenWithFormatting, FormattingKind} from "./../../manipulation";
+import {Node} from "./../common";
+import {NamedNode} from "./../base";
+
+export type AwaitableNodeExtensionType = Node<ts.Node & { awaitModifier?: ts.AwaitKeywordToken; }>;
+
+export interface AwaitableNode {
+    /**
+     * If it's a generator function.
+     */
+    isAwait(): boolean;
+    /**
+     * Gets the await token or undefined if none exists.
+     */
+    getAwaitKeywordToken(): Node<ts.AwaitKeywordToken> | undefined;
```

Comment:
I think we should drop `Token` here—`getAwaitKeyword` and `getAwaitKeywordOrThrow`.

Other places like `ReadonlyableNode` will use `getReadonlyKeyword()` instead of including `Token` (perhaps that should be `getReadonlyToken()`, but not a big deal).

Actually, it's kind of strange that in the compiler API, the only token with the word `Keyword` in it is `AwaitKeywordToken`.